addons/hr_holidays/tests/test_company_leave.py

```diff
@@ -334,3 +334,104 @@ def test_leave_whole_company_07(self):
 
         leaves = self.env['hr.leave'].search([('holiday_status_id', '=', self.bank_holiday.id)])
         self.assertEqual(len(leaves), 102)
+
+    def test_leave_whole_company_08(self):
+        """
+        Give a company leave with employees on different schedules.
+        """
+        # employee on different schedule
+        calendar = self.env['resource.calendar'].create({
+            'name': 'Different schedule',
+            'attendance_ids': [(5, 0, 0),
+                               (0, 0, {
+                                   'name': 'monday morning, earlier start',
+                                   'hour_from': 7.5,
+                                   'hour_to': 9.5,
+                                   'day_period': 'morning',
+                                   'dayofweek': '0',
+                               }),
+                               (0, 0, {
+                                   'name': 'monday morning, second attendance',
+                                   'hour_from': 10,
+                                   'hour_to': 12,
+                                   'day_period': 'morning',
+                                   'dayofweek': '0',
+                               })]
+        })
+        self.employee.resource_calendar_id = calendar
+
+        # employee on default schedule
+        self.employee2 = self.env['hr.employee'].create({
```

Comment:
This shouldn't write on self
```suggestion
        employee2 = self.env['hr.employee'].create({
```